File: app-ios/VelaWallet/VelaWallet/Features/Settings/CurrencyCatalog.swift
//
//  CurrencyCatalog.swift
//  VelaWallet
//
//  Which currencies the picker offers, and what each one is called.
//
//  Not the core's, deliberately. `display_currency` owns the *choice* and
//  whether a code can be priced right now; the LIST is provider-driven — the FX
//  endpoint decides what it can quote — so the names are data here rather than
//  120 corpus strings.
//
//  Shared by `SettingsFixtures` and `SettingsLive` since spec 050, for the same
//  reason `ContactsLabels` is: the gallery board and the live screen must offer
//  the same eight currencies with the same symbols, and two copies of that list
//  is how they stop.
//
//  Until 051 owns the rate path this is the eight the drawing shows. A
//  provider-driven list replaces the constant, not the shape.
//

import Foundation

enum CurrencyCatalog {

    struct Entry {
        let code: String
        /// The symbol the picker's leading badge draws. Not unique — HKD and
        /// USD share `$`, CNY and JPY share `¥` — which is why the code is the
        /// identifier and this is only decoration.
        let glyph: String
        /// Provider data, not corpus copy.
        let name: String
    }

    static let entries: [Entry] = [
        Entry(code: "USD", glyph: "$", name: "US Dollar"),
        Entry(code: "EUR", glyph: "€", name: "Euro"),
        Entry(code: "GBP", glyph: "£", name: "British Pound"),
        Entry(code: "CNY", glyph: "¥", name: "Chinese Yuan"),
        Entry(code: "JPY", glyph: "¥", name: "Japanese Yen"),
        Entry(code: "KRW", glyph: "₩", name: "South Korean Won"),
        Entry(code: "HKD", glyph: "$", name: "Hong Kong Dollar"),
        Entry(code: "VND", glyph: "₫", name: "Vietnamese Dong"),
    ]

    /// `nil` for a code the catalog does not know — which is a real case: the
    /// core will happily carry a code this list has never heard of, because the
    /// person's device region chose it.
    static func entry(_ code: String) -> Entry? {
        entries.first { $0.code == code }
    }
}

File: app-android/vela-wallet/app/src/main/java/app/getvela/wallet/feature/settings/core/CurrencyCatalog.kt
package app.getvela.wallet.feature.settings.core

/**
 * The currencies this wallet offers, and the glyph each one shows.
 *
 * **Shell content, not a core rule** — and deliberately so. The
 * `display_currency` machine decides *which* code is current and whether it
 * can be priced; it has no opinion about what a `¥` looks like or which eight
 * currencies a picker is worth filling with. The core's own comment says as
 * much: "the shell derives the symbol from its catalog and owns all
 * formatting".
 *
 * It lives here rather than in `SettingsFixtures` because live code must not
 * import from a fixture file: a fixture is a picture of a screen, and the
 * moment production reads one, deleting a picture breaks the product. Same
 * list, same order as `app-web/vela-wallet/src/lib/settings/fixtures.ts`, so
 * the two clients offer a person the same choices.
 */
object CurrencyCatalog {

    data class Entry(val code: String, val glyph: String, val caption: String)

    /**
     * Eight, in this order.
     *
     * The captions are English here and Chinese in the web sibling, which is a
     * real inconsistency — neither is localised — recorded as a debt in this
     * feature's results rather than fixed silently, because fixing it means
     * choosing i18n keys for currency names and that is a content decision.
     */
    val entries: List<Entry> = listOf(
        Entry("USD", "$", "US Dollar"),
        Entry("EUR", "€", "Euro"),
        Entry("GBP", "£", "British Pound"),
        Entry("CNY", "¥", "Chinese Yuan"),
        Entry("JPY", "¥", "Japanese Yen"),
        Entry("KRW", "₩", "South Korean Won"),
        Entry("HKD", "$", "Hong Kong Dollar"),
        Entry("VND", "₫", "Vietnamese Dong"),
    )

    /** The glyph for a code, or the code itself when it is one we do not stock. */
    fun glyph(code: String): String =
        entries.firstOrNull { it.code == code }?.glyph ?: code
}
